coordinator/src/collaborative_revert.rs

```diff
@@ -347,159 +164,50 @@ pub fn confirm_collaborative_revert(
-                format!("Could not get channel keys ID for subchannel {channel_id_hex}")
-            })?,
-    };
-
-    let own_sig = {
-        let fund_value_satoshis = funding_txo.value;
-
-        let signer = node
-            .keys_manager
-            .derive_ln_dlc_channel_signer(fund_value_satoshis, channel_keys_id);
-
-        signer
-            .get_holder_split_tx_signature(
-                &Secp256k1::new(),
-                &revert_transaction,
-                &funding_redeemscript,
-                fund_value_satoshis,
-            )
-            .context("Could not get own signature for collaborative revert transaction")?
-    };
+        .list_signed_dlc_channels()
+        .context("Failed to list dlc channels")?;
+    let signed_channel = subchannels
+        .iter()
+        .find(|c| c.channel_id == channel_id)
+        .context("DLC channel to be reverted not found")?;
 
-    let position = Position::get_position_by_trader(conn, record.trader_pubkey, vec![])?
-        .with_context(|| format!("Could not load position for subchannel {channel_id_hex}"))?;
+    let fund_out_amount = signed_channel.fund_tx.output[signed_channel.fund_output_index].value;
 
-    dlc::util::finalize_multi_sig_input_transaction(
-        &mut revert_transaction,
-        vec![
-            (own_fund_pk, own_sig),
-            (counter_fund_pk, counterparty_signature),
-        ],
-        &funding_redeemscript,
+    let own_fund_sk = node
+        .wallet()
+        .get_secret_key_for_pubkey(&signed_channel.own_params.fund_pubkey)?;
+
+    let mut close_tx = revert_transaction.clone();
+
+    dlc::util::sign_multi_sig_input(
+        &Secp256k1::new(),
+        &mut close_tx,
+        &counterparty_signature,
+        &signed_channel.counter_params.fund_pubkey,
+        &own_fund_sk,
+        &signed_channel.fund_script_pubkey,
+        fund_out_amount,
         0,
-    );
+    )?;
 
     tracing::info!(
-        txid = revert_transaction.txid().to_string(),
+        txid = close_tx.txid().to_string(),
         "Broadcasting collaborative revert transaction"
     );
     node.ldk_wallet()
-        .broadcast_transaction(&revert_transaction)
+        .broadcast_transaction(&close_tx)
         .context("Could not broadcast transaction")?;
 
     // TODO: We should probably not modify the state until the transaction has been confirmed.
 
+    let position = Position::get_position_by_trader(conn, record.trader_pubkey, vec![])?
+        .with_context(|| format!("Could not load position for subchannel {channel_id_hex}"))?;
+
     Position::set_position_to_closed(conn, position.id)
         .context("Could not set position to closed")?;
 
-    if let Some(mut subchannel) = optional_subchannel.cloned() {
-        subchannel.state = SubChannelState::OnChainClosed;
-        node.dlc_manager
-            .get_store()
-            .upsert_sub_channel(&subchannel)?;
-    }
-
     db::collaborative_reverts::delete(conn, channel_id)?;
 
```

Comment:
🔧 It would be nice to also cleanup the dlc channel state on the coordinator.

```suggestion
node.dlc_manager
        .get_store()
        .upsert_channel(
            dlc_manager::channel::Channel::CollaborativelyClosed(ClosedChannel {
                counter_party: signed_channel.counter_party,
                temporary_channel_id: signed_channel.temporary_channel_id,
                channel_id: signed_channel.channel_id,
            }),
            // The contract doesn't matter anymore
            None,
        )
        .map_err(|e| anyhow!("{e:#}"))
```

Reply:
I thought I did this? ... thanks for pointing it out.